flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/service/FlinkService.java

```diff
@@ -431,11 +431,12 @@ public void cancelJob(FlinkDeployment deployment, UpgradeMode upgradeMode) throw
         }
         deploymentStatus.getJobStatus().setState(JobStatus.FINISHED.name());
         savepointOpt.ifPresent(
-                location ->
-                        deploymentStatus
-                                .getJobStatus()
-                                .getSavepointInfo()
-                                .setLastSavepoint(Savepoint.of(location)));
+                location -> {
+                    Savepoint sp = Savepoint.of(location);
+                    deploymentStatus.getJobStatus().getSavepointInfo().setLastSavepoint(sp);
+                    // this is required for Flink < 1.15
+                    deploymentStatus.getJobStatus().getSavepointInfo().addSavepointToHistory(sp);
```

Comment:
We also need to do this for `FlinkSessionJobReconciler`.

Reply:
done!